File: SwarmPortal.Context/Models/Link.cs
using System.Diagnostics.CodeAnalysis;
using Microsoft.EntityFrameworkCore;

namespace SwarmPortal.Context;

[Index(nameof(Name), IsUnique = true)]
public class Link : ILink
{
    [NotNull]
    public ulong Id { get; set; }
    [NotNull]
    public string Name { get; set; } = null!;
    [NotNull]
    public string Url { get; set; } = null!;
    public Group Group { get; set; } = null!;
    public bool Enabled { get; set; }
    public ICollection<Role> Roles { get; set; } = null!;
    IEnumerable<IRole> ILink.Roles  => Roles;
}

File: SwarmPortal.React/src/services/openapi/models/Link.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Group } from './Group';
import type { Role } from './Role';

export type Link = {
    id?: number;
    name?: string | null;
    url?: string | null;
    group?: Group;
    enabled?: boolean;
    roles?: Array<Role> | null;
};

